### PR TITLE
chore: Add docs

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

- re-add linting rule
- add docs about the spectral updates
## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: .github/workflows/spectral-lint.yml [hunk 1]
```diff
@@ -32,4 +32,4 @@ jobs:
         with:
           # Path to the OpenAPI spec files and openapi/v2.yaml
           file_glob: openapi/v2.yaml
-          spectral_ruleset: tools/spectral/.spectral.yaml
+          spectral_ruleset: tools/spectral/.spectral.yaml #If updated, need to update in MMS too.
```
Comment (andreaangiolillo) [inline .github/workflows/spectral-lint.yml:35]: can we add a link to a PR that updates the MMS logic?

File: tools/spectral/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+If adding new rules or updating .spectral.yaml overall, the validations will instantly get updated across mongodb/openapi repository.
+
+To propagate the changes in MMS, engineers must open a PR and update the pinned commit sha in mms.
+
+Please perform the following steps:
+1. [ ] Open a PR with the spectral changes and validate that the Spectral lint checks work.
+2. [ ] Review and merge the PR.
+3. [ ] Open a PR in mms, updating the commit sha of the spectral file imported.
+4. [ ] Validate all tests pass.
```
Comment (andreaangiolillo) [inline tools/spectral/README.md:11]: same here to add an example of a PR to update mms or a link to the file to update

Comment (blva) [reply tools/spectral/README.md:11]: I was trying not to link MMS repo to public repo, do we know if that's OK?

Comment (andreaangiolillo) [reply tools/spectral/README.md:11]: ah good point 👍

**Review by andreaangiolillo — APPROVED** [review]

**Pull request merged**